dataset.py

```diff
@@ -19,20 +19,42 @@ def __len__(self):
         return self._length
 
     def __getitem__(self, index):
-        with self._reader.begin() as txn:
-            value = txn.get(self._keys[index])
+        if isinstance(index, slice):
```

Comment:
don't really understand this if statement, `__getitem__` should return only one data point (i.e. data at index) so why are you returning every single datapoint -> `result.append(image, length, digits, path)`?

Reply:
`__getitem__` can also used to implement slice in this way

Reply:
so you can return the whole dataset in __getitem__?

Reply:
It doesn't return the whole dataset, it will return based on indices of the `slice`, since slice has an `index.start` and `index.stop`